superset-frontend/src/components/DynamicEditableTitle/index.tsx

```diff
@@ -113,10 +114,7 @@ export const DynamicEditableTitle = ({
   // then we can measure the width of that span to resize the input element
   useLayoutEffect(() => {
     if (sizerRef?.current) {
-      sizerRef.current.innerHTML = (currentTitle || placeholder).replace(
-        /\s/g,
-        '&nbsp;',
-      );
+      sizerRef.current.innerHTML = escapeHTML(currentTitle || placeholder);
```

Comment:
This is a good improvement, yet it's a breaking change, minimal we need a line on `UPDATING.md`